insights/archive_functions.py

```diff
@@ -0,0 +1,26 @@
+import re
+import shutil
+import tarfile
+
+
+def get_system_id(self):
+    """ Checking system id from archive """
+    tar = tarfile.open(self.archive_location, 'r')
+    tar.extractall()
+    self.machine_id = tar.extractfile(self.machine_id_path)
```

Comment:
@kanchan04katare try to pass the parameters to the function like this:
def get_system_id(machine_id, archive_location)
While making a call as well to the above defined functions, respective parameters should be passed around. 
Refer -> https://github.com/redhataccess/rhi-pool/blob/master/insights/subset_functions.py#L3 for more clarity.